cloud-media-livestream/keypublisher/README.md

```diff
@@ -0,0 +1,302 @@
+gcloud alpha services api-keys create --api-target=service=[managedService]
+```
+
+You will get the `keyString` field in the response. This is your API key. Save
+this key, as you will not be able to retrieve it again.
+
+### Test your API
+
+Using the above information, use `curl` to make a `GET` query to your API:
+
+```shell
+curl --location --request POST \
+  'https://livestream-key-publisher-api-gateway-48mwfke9.uc.gateway.dev/keys?api_key=YOUR_API_KEY' \
+  --header 'Content-Type: application/json' \
+  --data '{"mediaId": "my-asset", "provider": "FakeProvider", "keyIds": ["abcd1234", "efgi5678"]}'
+```
+
+Arguments in the request body are:
+
+*   `mediaId`: arbitrary identifier for the media being encrypted.
+*   `provider`: the DRM provider to use. Omit this argument to see a list of
+    supported providers.
+*   `keyIds`: a list of key IDs to prepare for the given media ID.
+
+This example query uses `FakeProvider`. This is a provider used as a reference
+example, which does not make any actual CPIX queries, but generates random hex
+strings in place of a real key. These keys will not work for a real encryption
+setup.
+
+If the query is successful, you will see a response like:
+
+```
+projects/PROJECT_NUMBER/secrets/MEDIA_ID/versions/1
+```
+
+This is the name of the secret version that was created to hold your encryption
+keys.
+
+To verify this, you can use `gcloud` to access the content of that secret:
+
+```shell
+gcloud secrets versions access projects/PROJECT_NUMBER/secrets/MEDIA_ID/versions/1
+```
+
+## Update code
+
+Use `gcloud functions deploy` to update your Cloud Function code. Ensure
+`.env.yaml` still exists and its contents are correct.
+
+```shell
+gcloud functions deploy livestream-key-publisher \
+  --entry-point=keys \
+  --runtime=python310 \
+  --trigger-http \
+  --memory=256MB \
+  --security-level=secure-always \
+  --timeout=60s \
+  --env-vars-file=.env.yaml \
+  --service-account="livestream-cloud-functions@[MY-PROJECT].iam.gserviceaccount.com"
+```
+
+If request paths or parameters have changed, you may also need to update
+`api-config.yml` to match. To do this, update `api-config.yml` as needed, then
+create a new API config:
+
+```shell
+gcloud api-gateway api-configs create livestream-key-publisher-api-config-2 \
+  --display-name="Live Stream Key Publisher API Config v2" \
+  --api=livestream-key-publisher-api \
+  --openapi-spec="api-config.yml" \
+  --backend-auth-service-account="livestream-cf-invoker@[MY-PROJECT].iam.gserviceaccount.com"
+```
+
+Once the config has been created, update the API Gateway to use your new config:
+
+```shell
+gcloud api-gateway gateways update livestream-key-publisher-api-gateway \
+  --api=livestream-key-publisher-api \
+  --api-config=livestream-key-publisher-api-config-2 \
+  --location=us-central1
+```
+
+## Supported Python Versions
+
+The code samples for key publisher are compatible with all current active and
+maintenance versions of Python.
+
+Python >= 3.7
```

Comment:
I'd suggest linking to a page rather than specifying a version. 3.7 will also be end of life in about a month (June 27?)

Reply:
Done.